driver-core/src/main/com/mongodb/internal/DocumentToDBRefTransformer.java

```diff
@@ -14,12 +14,16 @@
  * limitations under the License.
  */
 
-package com.mongodb;
+package com.mongodb.internal;
```

Comment:
I suggest making this public, as I don't think it's a good precedent to have driver-async/driver depend on internal classes of driver-core (at least I think this would be setting a precedent).  It would also allow an application to easily construct DocumentCodecProvider instances that decode to DBRef